src/pages/resultsView/ResultsViewPage.tsx

```diff
@@ -89,94 +105,54 @@ function initStore() {
-
-                // filter out duplicates
-                molecularProfiles = _.uniq(molecularProfiles);
+                if (virtualStudies.length > 0) {
 
-                return molecularProfiles;
-            }
-
-            runInAction(() => {
+                    // we want to replace virtual studies with their underlying physical studies
+                    physicalStudies = substitutePhysicalStudiesForVirtualStudies(cancerStudyIds, virtualStudies);
 
-                if (!resultsViewPageStore._samplesSpecification || !_.isEqual(resultsViewPageStore._samplesSpecification.slice(), samplesSpecification)) {
-                    resultsViewPageStore._samplesSpecification = samplesSpecification;
-                }
+                    samplesSpecification = populateSampleSpecificationsFromVirtualStudies(samplesSpecification, virtualStudies);
 
-                // sometimes the submitted case_set_id is not actually a case_set_id but
-                // a category of case set ids (e.g. selected studies > 1 and case category selected)
-                // in that case, note that on the query
-                if (query.case_set_id && ["w_mut","w_cna","w_mut_cna"].includes(query.case_set_id)) {
-                    if (resultsViewPageStore.sampleListCategory !== query.case_set_id) {
-                        resultsViewPageStore.sampleListCategory = query.case_set_id;
-                    }
                 } else {
-                    resultsViewPageStore.sampleListCategory = undefined;
-                }
-
-                if (query.data_priority !== undefined && parseInt(query.data_priority,10) !== resultsViewPageStore.profileFilter) {
-                    resultsViewPageStore.profileFilter = parseInt(query.data_priority,10);
+                    physicalStudies = cancerStudyIds;
                 }
 
-                // note that this could be zero length if we have multiple studies
-                // in that case we derive default selected profiles
-                const profiles = getMolecularProfiles(query);
-                if (!resultsViewPageStore.selectedMolecularProfileIds || !_.isEqual(resultsViewPageStore.selectedMolecularProfileIds.slice(), profiles)) {
-                    resultsViewPageStore.selectedMolecularProfileIds = profiles;
-                }
+                // fetch summary projection because it's likely to be already used (an in cache) or will be used later
+                runInAction(()=>{
+                    updateStoreFromQuery(resultsViewPageStore, query, samplesSpecification, physicalStudies, oql, cancerStudyIds);
+                });
 
-                if (!_.isEqual(query.RPPA_SCORE_THRESHOLD, resultsViewPageStore.rppaScoreThreshold)) {
-                    resultsViewPageStore.rppaScoreThreshold = parseFloat(query.RPPA_SCORE_THRESHOLD);
-                }
+                // now update state so view can render result
+                resultsViewPageStore.checkingVirtualStudies = false;
```

Comment:
@adamabeshouse i hate this pattern here but am not sure yet how else to do this.

Reply:
Basically, we when we detect virtual studies in query, we need to fetch them and fixup the query specification.  The reaction makes this kind of awkward.    Maybe if instead of reaction we were getting "query" as a property, there would a more idiomatic way to to do this.

Reply:
what about making the fetch into a mobxpromise and using .isPending? you could still reference the result (and the status) in the reaction